bbl/bbl.c

```diff
@@ -40,6 +40,52 @@ static void filter_dtb(uintptr_t source)
   filter_compat(dest, "riscv,debug-013");
 }
 
+static void protect_memory(void)
+{
+  // Check to see if up to four PMP registers are implemented.
+  // Ignore the illegal-instruction trap if PMPs aren't supported.
+  uintptr_t a0 = 0, a1 = 0, a2 = 0, a3 = 0, tmp, cfg;
+  asm volatile ("la %[tmp], 1f\n\t"
+                "csrrw %[tmp], mtvec, %[tmp]\n\t"
```

Comment:
I think this is OK: it'll break if an m-mode interrupt comes in, but there shouldn't be any that can fire here because everything is delegated by hardware to S mode.

Reply:
In general, bbl/pk assume that interrupts are disabled while in M-mode (which is the case).